spark-screenshot-testing/src/test/kotlin/com/adevinta/spark/PaparazziUtils.kt

```diff
@@ -132,4 +147,37 @@ internal fun Paparazzi.sparkSnapshotNightMode(
     }
 }
 
+/**
+ * Generate 2 screenshots for high contrast themes: light and dark high contrast
+ */
+internal fun Paparazzi.sparkSnapshotHighContrast(
+    name: String? = null,
+    drawBackground: Boolean = true,
+    composable: @Composable () -> Unit,
+) {
+    var lightTextThrowable: Throwable? = null
+    HighContrastThemeVariant.entries.forEach {
+        try {
+            sparkSnapshotWithColors(
+                name = name.orEmpty() + "_${it.name}",
+                drawBackground = drawBackground,
+                colors = when (it) {
+                    HighContrastThemeVariant.LightHighContrast -> lightHighContrastSparkColors()
+                    HighContrastThemeVariant.DarkHighContrast -> darkHighContrastSparkColors()
+                },
+                composable = composable,
+            )
+        } catch (e: Throwable) {
```

Comment:
What was the initial reason for doing this custom logic instead of just "rethrowing the last caught exception"?

Reply:
Unfortunately, I don't remember 🫤, maybe to only have the light exception as they tend to be more readable I would say 🤔

I agree that this seems over-complicated for the usecase, I'll simplify it 🙂‍↕️